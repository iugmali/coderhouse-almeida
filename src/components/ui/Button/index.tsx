'use client';

import React from "react";

type Props = {
  children?: React.ReactNode;
  disabled?: boolean;
  className?: string;
  handleClick: () => void;
};

const Button = ({children, handleClick, disabled = false, className = ''}: Props) => {
  return (
    <button className={`${className} rounded bg-blue-900 disabled:bg-gray-500 px-4 py-2 text-white hover:opacity-90 active:opacity-70 disabled:hover:opacity-100`} onClick={handleClick} disabled={disabled}>
      {children}
    </button>
  );
}

export default Button;
