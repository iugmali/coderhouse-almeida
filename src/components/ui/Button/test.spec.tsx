import {render} from "@testing-library/react";
import Button, {TextButton} from "@/components/ui/Button/index";

describe('Button', () => {
  describe('default Button', () => {
    it('renders unchanged', () => {
      const {container} = render(<Button handleClick={jest.fn()}>button</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
  describe('text Button', () =>{
    it('renders unchanged', () => {
      const {container} = render(<TextButton handleClick={jest.fn()}>button</TextButton> )
      expect(container.firstChild).toMatchSnapshot();
    })
  });
});
