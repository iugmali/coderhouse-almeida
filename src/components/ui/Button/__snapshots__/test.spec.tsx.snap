// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default Button renders unchanged 1`] = `
<button
  class="rounded bg-blue-900 disabled:bg-gray-500 px-4 py-2 text-white hover:opacity-90 active:opacity-70 disabled:hover:opacity-100"
  style="opacity: 0; transform: scale(0) translateZ(0);"
  tabindex="0"
>
  button
</button>
`;

exports[`Button default Button should render properly 1`] = `
<button
  class="rounded bg-blue-900 disabled:bg-gray-500 px-4 py-2 text-white hover:opacity-90 active:opacity-70 disabled:hover:opacity-100"
  style="opacity: 0; transform: scale(0) translateZ(0);"
  tabindex="0"
>
  button
</button>
`;

exports[`Button text Button renders unchanged 1`] = `
<button
  class="rounded bg-gray-50 disabled:bg-gray-500 px-4 py-2 text-gray-950 disabled:hover:opacity-100"
  style="opacity: 0; transform: scale(0) translateZ(0);"
  tabindex="0"
>
  button
</button>
`;

exports[`Button text Button should render properly 1`] = `
<button
  class="rounded bg-gray-50 disabled:bg-gray-500 px-4 py-2 text-gray-950 disabled:hover:opacity-100"
  style="opacity: 0; transform: scale(0) translateZ(0);"
  tabindex="0"
>
  button
</button>
`;
